### PR TITLE
fix: bug preventing compiling multiple versions

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

fixes: #27 

### How I did it

Create a mapping of version to files before running through the  solcx

### How to verify it

Compile projects that use multiple Solidity versions.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ape_solidity/compiler.py [hunk 1]
```diff
@@ -184,58 +182,56 @@ def compile(
...
                 solc_version = max(self.installed_versions)
 
-        if not base_path:
-            # Does support base_path
-            from ape import config
+            if solc_version not in files_by_solc_version:
+                files_by_solc_version[solc_version] = set()
 
-            base_path = config.contracts_folder
+            files_by_solc_version[solc_version].add(path)
```
Comment (fubuloubu) [inline ape_solidity/compiler.py:189]: Add an assertion that we didn't lose any files by doing this (e.g. `set(contract_filepaths)` is the same as the aggregated set of all files by version)

**Review by fubuloubu — APPROVED** [review]

File: ape_solidity/compiler.py (hunk 1 above)
Comment (unparalleled-js) [reply ape_solidity/compiler.py:206]: I would rather cover this in tests!

Comment (unparalleled-js) [reply ape_solidity/compiler.py:206]: This is already covered in the tests really.

File: ape_solidity/compiler.py [hunk 2]
```diff
@@ -163,79 +163,100 @@ def compile(
         self, contract_filepaths: List[Path], base_path: Optional[Path] = None
     ) -> List[ContractType]:
         # TODO: move this to solcx
-        contract_types = []
-        files = []
-        solc_version = None
+        contract_types: List[ContractType] = []
+        files_by_solc_version: Dict[str, Set[Path]] = {}
+        solc_version_by_file_name: Dict[str, str] = {}
```
Comment (fubuloubu) [inline ape_solidity/compiler.py:168]: Can these be more specific like this?
```suggestion
        files_by_solc_version: Dict[Version, Set[Path]] = {}
        solc_version_by_file: Dict[Path, Version] = {}
```

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**